internal/servers/controller/controller.go

```diff
@@ -124,7 +124,8 @@ func New(ctx context.Context, conf *Config) (*Controller, error) {
 	}
 
 	clusterListeners := make([]*base.ServerListener, 0)
-	for _, l := range conf.Listeners {
+	for i := range conf.Listeners {
+		l := conf.Listeners[i]
```

Comment:
We're iterating like this quite a bit in this flow, but honestly it's mostly because I keep forgetting when I _need_ to do this to iterate over a slice of pointers 😂 
I don't think it's needed in here though.